apps/app/src/generator/GeneratorContext.tsx

```diff
@@ -86,14 +89,13 @@ const GeneratorProvider = ({ children }: { children: React.ReactNode }) => {
         if (!isBaseChange) {
           if (!isCodeEdit) {
             const diff = themeDiff(prev, changes);
-            return merge({}, prev, diff);
+            return mergeTheme(prev, diff);
```

Comment:
This doesn't seem to be working as expected: when I try to update a color and then another, the first one is removed from the custom theme

Reply:
`mergeTheme` was squashing the previous sub-object; I've added a fix